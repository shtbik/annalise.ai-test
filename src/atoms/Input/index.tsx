import { FC, InputHTMLAttributes } from 'react'
import { StyledComponent } from '../../types/component'

import { StyledInput } from './styled.index'

interface IInput extends StyledComponent {}

const Input: FC<IInput & InputHTMLAttributes<HTMLInputElement>> = ({ ...props }) => {
  return <StyledInput {...props} />
}

export default Input
