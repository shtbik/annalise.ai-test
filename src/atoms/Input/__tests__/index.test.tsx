import { render } from '@testing-library/react'
import styled from 'styled-components'

import Input from '../'

const SInput = styled(Input)`
  display: none;
`

describe('Input component', () => {
  it('pass classname', () => {
    const { container } = render(<SInput name="test" />)
    expect(container.firstChild).toHaveClass(SInput['styledComponentId'])
    expect(container.firstChild).toHaveStyleRule('display', 'none')
  })

  it('snapshot', () => {
    const { container } = render(<Input name="surname" placeholder="type your surname" defaultValue="Pupkin" />)
    expect(container).toMatchSnapshot()
  })
})
