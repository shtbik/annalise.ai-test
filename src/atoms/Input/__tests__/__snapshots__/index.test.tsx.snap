// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component snapshot 1`] = `
.c0 {
  padding: 5px;
}

<div>
  <input
    class="c0"
    name="surname"
    placeholder="type your surname"
    value="Pupkin"
  />
</div>
`;
